python/pyspark/pandas/frame.py

```diff
@@ -10239,8 +10239,10 @@ def any(self, axis: Axis = 0) -> "Series":
 
         return first_series(DataFrame(internal))
 
-    # TODO: add axis, numeric_only, pct, na_option parameter
-    def rank(self, method: str = "average", ascending: bool = True) -> "DataFrame":
+    # TODO: add axis, pct, na_option parameter
+    def rank(
+        self, method: str = "average", ascending: bool = True, numeric_only: Optional[bool] = None
```

Comment:
https://github.com/pandas-dev/pandas/blob/6033ed4b3383d874ee4a8a461724c0b8c2ca968d/pandas/core/generic.py#L8651-L8660

Consider rank=`None` would be deprecated in future in pandas, we might want to add a test on this (set `rank=None` expilictly), to make sure protect some behavior in future.

Reply:
Makes sense! Tests added.